### DISCUSSION TITLE
Moving toward single global VLAN namespace

### DISCUSSION THREAD

**Discussion by ipspace** [ipspace/netlab#639] [Features]
In theory, the current implementation of the VLAN module supports node-specific VLANs with the same name. For example, multiple nodes might have a **red** VLAN, but as long as it's not defined globally, these are not the same VLANs and will get different VLAN IDs (at least that's how I remember the code behaving 🥴).

It would make everyone's life easier if we'd go for globally unique VLAN namespace, disallowing having the same VLAN name on multiple nodes without having a global VLAN with the same name (or creating global VLAN on the fly). Obviously you could still use `red_s1`, `red_s2`, `red_s3` if you wish.

Please note that this change would not impact routed VLANs used in stuff like VRF Lite. Those VLANs always get an IP prefix per physical link.

So: is anyone using node-specific VLANs @jbemmel @ssasso @petercrocker?

**Comment by ssasso** [comment]
I have some use cases where I want to use the same vlan id on different links (but not the same vlan "globally speaking") - but never tested yet with netlab, even if it's in my todo list.
So, even if using a global vlan namespace, this should allow "overlapping" IDs...

So, if I can have
```
vlans:
  s1_red:
    id: 10
  s2_red:
    id: 10
```
for me it's ok.

**Reply by ipspace** [reply]
Oh, that wouldn't work. I think we check for unique IDs. But there's nothing enforcing VLAN continuity, so it's perfectly OK to have a bit of VLAN 10 here, another bit of VLAN 10 over there... as long as you're OK with VLAN 10 having the same IP prefix everywhere.

**Reply by jbemmel** [reply]
Netlab ties all ports with a given VLAN ID together into a single L2 domain. Building a model with 'port significant' vlans would only be possible using a plug-in

**Comment by ipspace** [comment]
@jbemmel fixed #287 -- this discussion is moot
